### ISSUE TITLE
Clearer error message for pd.concat if empty iterator is passed

### ISSUE THREAD

**Issue by cel4** [pydata/pandas#9157]
I just saw following problem on stackoverflow: http://stackoverflow.com/questions/27665560/concatenating-files-with-python-pandas

It is not possible to concatenate an empty sequence of data frames.

This tiny code fragment
```
import pandas as pd
dfs = []
pd.concat(dfs)
``` 

Throws following error:
```
/Users/ch/miniconda/envs/sci33/lib/python3.3/site-packages/pandas/tools/merge.py in __init__(self, objs, axis, join, join_axes, keys, levels, names, ignore_index, verify_integrity, copy)
    765 
    766         if len(objs) == 0:
--> 767             raise ValueError('All objects passed were None')
    768 
    769         # consolidate data & figure out what our result ndim is going to be

ValueError: All objects passed were None
```

I would suggest catching this special case in `pd.concat` before trying to merge and to raise an error with a clearer error description. For this special case, the current error message is not very helpful.

**Comment by springcoil** [comment]
Is this one closed or open still?

**Comment by cel4** [comment]
@springcoil, thanks for the reminder. This should've been closed along with my closed pr. Now my PR is open again and I am finishing it, so this issue will hopefully be merged-closed soon.

**Comment by jreback** [comment]
no, that's the point of closing on merging of a PR and not before

**Comment by jreback** [comment]
closed by #9172

**Comment by kishore0401** [comment]
Is there a way to return an empty dataframe with by-passing the error message
